android/src/main/java/com/lykhonis/flutterbilling/BillingPlugin.java

```diff
@@ -185,6 +231,48 @@ public void failed() {
         });
     }
 
+    private void fetchSubscriptions(final List<String> identifiers, final Result result) {
+        executeServiceRequest(new Request() {
+            @Override
+            public void execute() {
+                billingClient.querySkuDetailsAsync(
+                        SkuDetailsParams.newBuilder()
+                                        .setSkusList(identifiers)
+                                        .setType(SkuType.SUBS)
+                                        .build(),
+                        new SkuDetailsResponseListener() {
+                            @Override
+                            public void onSkuDetailsResponse(int responseCode, List<SkuDetails> skuDetailsList) {
+                                if (responseCode == BillingResponse.OK) {
+                                    final List<Map<String, Object>> products = new ArrayList<>();
+
+                                    for (SkuDetails details : skuDetailsList) {
+                                        final Map<String, Object> product = new HashMap<>();
```

Comment:
This section seems to be duplicated code, let's have some private function that maps details to a HashMap and use it here and in original function for products fetching